### ISSUE TITLE
CLIP contrastive image-text inference can't run on hpu with gaudi-docker/1.17.0

### ISSUE THREAD

**Issue by caijimin** [huggingface/optimum-habana#1216]
### System Info

```shell
hl-smi
+-----------------------------------------------------------------------------+
| HL-SMI Version:                                hl-1.17.0-fw-51.3.0          |
| Driver Version:                                     1.17.0-8a5dfb8          |
|-------------------------------+----------------------+----------------------+

docker vault.habana.ai/gaudi-docker/1.17.0/ubuntu22.04/habanalabs/pytorch-installer-2.3.1:latest
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
cd examples/contrastive-image-text
python run_clip.py    \
   --output_dir ./clip-roberta-finetuned \
   --model_name_or_path ./clip-roberta \
   --data_dir /DISK1/data/datasets/COCO  \
   --dataset_name ydshieh/coco_dataset_script   \
   --dataset_config_name=2017    \
   --image_column image_path    \
   --caption_column caption  \
   --remove_unused_columns=False   \
   --do_eval  \
   --per_device_eval_batch_size="64" 
   --overwrite_output_dir   \
   --use_habana  \
   --use_lazy_mode  \
   --use_hpu_graphs_for_inference  \
   --gaudi_config_name Habana/clip   \
   --bf16   \
   --mediapipe_dataloader  \
   --trust_remote_code true
```
```
 HABANA PT BRIDGE CONFIGURATION
 PT_HPU_LAZY_MODE = 1
 PT_RECIPE_CACHE_PATH = 
 PT_CACHE_FOLDER_DELETE = 0
 PT_HPU_RECIPE_CACHE_CONFIG = 
 PT_HPU_MAX_COMPOUND_OP_SIZE = 9223372036854775807
 PT_HPU_LAZY_ACC_PAR_MODE = 1
 PT_HPU_ENABLE_REFINE_DYNAMIC_SHAPES = 0
---------------------------: System Configuration :---------------------------
Num CPU Cores : 112
CPU RAM       : 1056426836 KB
------------------------------------------------------------------------------
 Warning!!!!!! : Unsupported device please use legacy/cpu/mixed
Falling back to legacy
MediaPipe device legacy device_type legacy device_id 0 pipe_name ClipMediaPipe:0
08/07/2024 08:42:44 - WARNING - clip_mediapipe_dataloader - Using Pytorch native dataloader because: ('ImageDecoder', ' sent with invalid device ', 'GAUDI2').
```
Inference will run on CPU, not on HPU.

I also tried Whisper, 
```
python run_speech_recognition_seq2seq.py \
    --model_name_or_path=/DISK1/data/models/whisper-small \
    --dataset_name="mozilla-foundation/common_voice_11_0" \
    --dataset_config_name="hi" \
    --language="hindi" \
    --task="transcribe" \
    --eval_split_name="test" \
    --gaudi_config_name="Habana/whisper" \
    --output_dir="./results/whisper-small-clean" \
    --per_device_eval_batch_size="32" \
    --generation_max_length="225" \
    --preprocessing_num_workers="1" \
    --max_duration_in_seconds="30" \
    --text_column_name="sentence" \
    --freeze_feature_encoder="False" \
    --bf16 \
    --overwrite_output_dir \
    --do_eval \
    --predict_with_generate \
    --use_habana \
    --use_hpu_graphs_for_inference \
    --label_features_max_length 128 \
    --dataloader_num_workers 8 \
    --trust_remote_code true	
```
Although it doesn't report error, it also runs on CPU instead of HPU.

BTW: I tried same script with `vault.habana.ai/gaudi-docker/1.16.2/ubuntu22.04/habanalabs/pytorch-installer-2.2.2:latest`, it runs well on HPU.

### Expected behavior

Clip and Whisper inference should run on HPU.

**Comment by regisss** [comment]
Can you try again now that #1163 has been merged?

**Comment by caijimin** [comment]
> Can you try again now that #1163 has been merged?

Thanks so much for your quick guidance.
It runs well with latest code, and need use  latest main branch 
```
pip install git+https://github.com/huggingface/optimum-habana.git
```

**Comment by caijimin** [comment]
Clip runs well but seems Whisper is still runs on CPU.

**Comment by regisss** [comment]
Can you share the command you ran please?

**Comment by caijimin** [comment]
The same as https://github.com/huggingface/optimum-habana/tree/main/examples/speech-recognition#single-hpu-seq2seq-inference
```
python run_speech_recognition_seq2seq.py \
    --model_name_or_path=/DISK1/data/models/whisper-small \
    --dataset_name="mozilla-foundation/common_voice_11_0" \
    --dataset_config_name="hi" \
    --language="hindi" \
    --task="transcribe" \
    --eval_split_name="test" \
    --gaudi_config_name="Habana/whisper" \
    --output_dir="./results/whisper-small-clean" \
    --per_device_eval_batch_size="32" \
    --generation_max_length="225" \
    --preprocessing_num_workers="1" \
    --max_duration_in_seconds="30" \
    --text_column_name="sentence" \
    --freeze_feature_encoder="False" \
    --bf16 \
    --overwrite_output_dir \
    --do_eval \
    --predict_with_generate \
    --use_habana \
    --use_hpu_graphs_for_inference \
    --label_features_max_length 128 \
    --dataloader_num_workers 8 \
    --trust_remote_code true	
```
BTW: Fine tune can run on HPU as https://github.com/huggingface/optimum-habana/tree/main/examples/speech-recognition#single-hpu-whisper-fine-tuning-with-seq2seq

**Comment by caijimin** [comment]
> BTW: Fine tune can run on HPU as https://github.com/huggingface/optimum-habana/tree/main/examples/speech-recognition#single-hpu-whisper-fine-tuning-with-seq2seq

Training can run on HPU as follows
```
[INFO|trainer.py:785] 2024-08-08 08:12:29,325 >> ***** Running training *****
[INFO|trainer.py:786] 2024-08-08 08:12:29,325 >>   Num examples = 4,361
[INFO|trainer.py:787] 2024-08-08 08:12:29,325 >>   Num Epochs = 3
[INFO|trainer.py:788] 2024-08-08 08:12:29,325 >>   Instantaneous batch size per device = 8
[INFO|trainer.py:791] 2024-08-08 08:12:29,325 >>   Total train batch size (w. parallel, distributed & accumulation) = 8
[INFO|trainer.py:792] 2024-08-08 08:12:29,325 >>   Gradient Accumulation steps = 1
[INFO|trainer.py:793] 2024-08-08 08:12:29,325 >>   Total optimization steps = 1,638
[INFO|trainer.py:794] 2024-08-08 08:12:29,329 >>   Number of trainable parameters = 240,582,912
```
but evaluation will run on CPU
```
***** Running Evaluation *****
[INFO|trainer.py:1832] 2024-08-08 08:15:06,144 >>   Num examples = 2894
[INFO|trainer.py:1835] 2024-08-08 08:15:06,144 >>   Batch size = 32
```

**Comment by regisss** [comment]
How do you notice it runs on CPU? It's slower?

**Comment by caijimin** [comment]
The utilization of HPU is always 0
```
# hl-smi 
+-----------------------------------------------------------------------------+
| HL-SMI Version:                                hl-1.17.0-fw-51.3.0          |
| Driver Version:                                     1.17.0-8a5dfb8          |
|-------------------------------+----------------------+----------------------+
| AIP  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | AIP-Util  Compute M. |
|===============================+======================+======================|
|   0  HL-225D             N/A  | 0000:48:00.0     N/A |                   0  |
| N/A   27C   N/A    96W / 450W |    768MiB / 98304MiB |     0%           N/A |
|-------------------------------+----------------------+----------------------+
|   1  HL-225D             N/A  | 0000:c2:00.0     N/A |                   0  |
| N/A   26C   N/A    71W / 450W |  23912MiB / 98304MiB |     0%           N/A |
|-------------------------------+----------------------+----------------------+
| Compute Processes:                                               AIP Memory |
|  AIP       PID   Type   Process name                             Usage      |
|=============================================================================|
|   0        N/A   N/A    N/A                                      N/A        |
|   1        4438     C   pt_main_thread                          23144MiB    |
+=============================================================================+
```
CPU usage is very high
```                                         
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                            
   4438 root      20   0   17.5g   3.9g 995.9m R  5725   0.4 642:52.96 pt_main_thread  
```
The output of inference scripts is
```
============================= HABANA PT BRIDGE CONFIGURATION =========================== 
 PT_HPU_LAZY_MODE = 1
 PT_RECIPE_CACHE_PATH = 
 PT_CACHE_FOLDER_DELETE = 0
 PT_HPU_RECIPE_CACHE_CONFIG = 
 PT_HPU_MAX_COMPOUND_OP_SIZE = 9223372036854775807
 PT_HPU_LAZY_ACC_PAR_MODE = 1
 PT_HPU_ENABLE_REFINE_DYNAMIC_SHAPES = 0
---------------------------: System Configuration :---------------------------
Num CPU Cores : 112
CPU RAM       : 1056426836 KB
------------------------------------------------------------------------------
08/09/2024 02:09:18 - INFO - __main__ - *** Evaluate ***
[INFO|trainer.py:811] 2024-08-09 02:09:18,343 >> The following columns in the evaluation set don't have a corresponding argument in `WhisperForConditionalGeneration.forward` and have been ignored: input_length. If input_length are not expected by `WhisperForConditionalGeneration.forward`,  you can safely ignore this message.
/usr/local/lib/python3.10/dist-packages/habana_frameworks/torch/hpu/__init__.py:160: UserWarning: torch.hpu.setDeterministic is deprecated and will be removed in next release. Please use torch.use_deterministic_algorithms instead.
  warnings.warn(
[2024-08-09 02:09:18,360] [INFO] [real_accelerator.py:191:get_accelerator] Setting ds_accelerator to hpu (auto detect)
[INFO|trainer.py:1812] 2024-08-09 02:09:18,578 >> Using HPU graphs for inference.
[INFO|trainer.py:1830] 2024-08-09 02:09:18,579 >> 
***** Running Evaluation *****
[INFO|trainer.py:1832] 2024-08-09 02:09:18,580 >>   Num examples = 2894
[INFO|trainer.py:1835] 2024-08-09 02:09:18,580 >>   Batch size = 32
[INFO|generation_whisper.py:1655] 2024-08-09 02:09:23,530 >> Increase max_length from 225 to 228 since input is conditioned on previous segment.
[WARNING|logging.py:328] 2024-08-09 02:09:23,588 >> The attention mask is not set and cannot be inferred from input because pad token is same as eos token.As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
[WARNING|logging.py:328] 2024-08-09 02:09:25,977 >> Passing a tuple of `past_key_values` is deprecated and will be removed in Transformers v4.43.0. You should pass an instance of `EncoderDecoderCache` instead, e.g. `past_key_values=EncoderDecoderCache.from_legacy_cache(past_key_values)`.
  0%|                                                                                                       | 0/91 [00:00<?, ?it/s][INFO|generation_whisper.py:1655] 2024-08-09 02:14:09,102 >> Increase max_length from 225 to 228 since input is conditioned on previous segment.
  2%|██                                                                                           | 2/91 [02:05<1:33:05, 62.75s/it][INFO|generation_whisper.py:1655] 2024-08-09 02:16:14,611 >> Increase max_length from 225 to 228 since input is conditioned on previous segment.
```

**Comment by caijimin** [comment]
I removed `--predict_with_generate ` and it will run on HPU.

**Comment by caijimin** [comment]
For Clip training with DeepSpeed, the loss is `nan`
```
python ../gaudi_spawn.py --world_size 8 --use_deepspeed run_clip.py  \
    --output_dir ./clip-roberta-finetuned     \
    --model_name_or_path ./clip-roberta     \
    --data_dir /DISK1/data/datasets/COCO   \
    --dataset_name ydshieh/coco_dataset_script \
    --dataset_config_name=2017   \
    --image_column image_path \
    --caption_column caption   \
    --remove_unused_columns=False \
    --do_train  --do_eval \
    --per_device_train_batch_size="512"   \
    --per_device_eval_batch_size="64"   \
    --learning_rate="5e-5" --warmup_steps="0" --weight_decay 0.1  \
    --overwrite_output_dir     --save_strategy epoch   \
     --use_habana     --use_lazy_mode  \
     --use_hpu_graphs_for_inference   \
     --gaudi_config_name Habana/clip 
    --throughput_warmup_steps 3  \
    --deepspeed ../../tests/configs/deepspeed_zero_2.json\
    --trust_remote_code true \
    --mediapipe_dataloader \
    --use_hpu_graphs_for_training

============================= HABANA PT BRIDGE CONFIGURATION =========================== 
 PT_HPU_LAZY_MODE = 1
 PT_RECIPE_CACHE_PATH = 
 PT_CACHE_FOLDER_DELETE = 0
 PT_HPU_RECIPE_CACHE_CONFIG = 
 PT_HPU_MAX_COMPOUND_OP_SIZE = 9223372036854775807
 PT_HPU_LAZY_ACC_PAR_MODE = 0
 PT_HPU_ENABLE_REFINE_DYNAMIC_SHAPES = 1
---------------------------: System Configuration :---------------------------
Num CPU Cores : 112
CPU RAM       : 1056426836 KB
------------------------------------------------------------------------------

[2024-08-12 07:41:07,170] [INFO] [logging.py:96:log_dist] [Rank 0] DeepSpeed Basic Optimizer = FusedAdamW
[2024-08-12 07:41:07,171] [INFO] [utils.py:56:is_zero_supported_optimizer] Checking ZeRO support for optimizer=FusedAdamW type=<class 'habana_frameworks.torch.hpex.optimizers.FusedAdamW.FusedAdamW'>
[2024-08-12 07:41:07,171] [WARNING] [engine.py:1196:_do_optimizer_sanity_check] **** You are using ZeRO with an untested optimizer, proceed with caution *****
[2024-08-12 07:41:07,170] [INFO] [logging.py:96:log_dist] [Rank 0] DeepSpeed Basic Optimizer = FusedAdamW
[2024-08-12 07:41:07,171] [INFO] [utils.py:56:is_zero_supported_optimizer] Checking ZeRO support for optimizer=FusedAdamW type=<class 'habana_frameworks.torch.hpex.optimizers.FusedAdamW.FusedAdamW'>
[2024-08-12 07:41:07,171] [WARNING] [engine.py:1196:_do_optimizer_sanity_check] **** You are using ZeRO with an untested optimizer, proceed with caution *****


***** train metrics *****
  epoch                       =         3.0
  max_memory_allocated (GB)   =       48.81
  memory_allocated (GB)       =       47.06
  total_flos                  = 221557680GF
  total_memory_available (GB) =       93.55
  train_loss                  =         nan
  train_runtime               =  0:04:03.07
  train_samples_per_second    =    7888.147
  train_steps_per_second      =       1.933

```

**Comment by xinyu-intel** [comment]
> For Clip training with DeepSpeed, the loss is `nan`
> 
> ```
> python ../gaudi_spawn.py --world_size 8 --use_deepspeed run_clip.py  \
>     --output_dir ./clip-roberta-finetuned     \
>     --model_name_or_path ./clip-roberta     \
>     --data_dir /DISK1/data/datasets/COCO   \
>     --dataset_name ydshieh/coco_dataset_script \
>     --dataset_config_name=2017   \
>     --image_column image_path \
>     --caption_column caption   \
>     --remove_unused_columns=False \
>     --do_train  --do_eval \
>     --per_device_train_batch_size="512"   \
>     --per_device_eval_batch_size="64"   \
>     --learning_rate="5e-5" --warmup_steps="0" --weight_decay 0.1  \
>     --overwrite_output_dir     --save_strategy epoch   \
>      --use_habana     --use_lazy_mode  \
>      --use_hpu_graphs_for_inference   \
>      --gaudi_config_name Habana/clip 
>     --throughput_warmup_steps 3  \
>     --deepspeed ../../tests/configs/deepspeed_zero_2.json\
>     --trust_remote_code true \
>     --mediapipe_dataloader \
>     --use_hpu_graphs_for_training
> 
> ============================= HABANA PT BRIDGE CONFIGURATION =========================== 
>  PT_HPU_LAZY_MODE = 1
>  PT_RECIPE_CACHE_PATH = 
>  PT_CACHE_FOLDER_DELETE = 0
>  PT_HPU_RECIPE_CACHE_CONFIG = 
>  PT_HPU_MAX_COMPOUND_OP_SIZE = 9223372036854775807
>  PT_HPU_LAZY_ACC_PAR_MODE = 0
>  PT_HPU_ENABLE_REFINE_DYNAMIC_SHAPES = 1
> ---------------------------: System Configuration :---------------------------
> Num CPU Cores : 112
> CPU RAM       : 1056426836 KB
> ------------------------------------------------------------------------------
> 
> [2024-08-12 07:41:07,170] [INFO] [logging.py:96:log_dist] [Rank 0] DeepSpeed Basic Optimizer = FusedAdamW
> [2024-08-12 07:41:07,171] [INFO] [utils.py:56:is_zero_supported_optimizer] Checking ZeRO support for optimizer=FusedAdamW type=<class 'habana_frameworks.torch.hpex.optimizers.FusedAdamW.FusedAdamW'>
> [2024-08-12 07:41:07,171] [WARNING] [engine.py:1196:_do_optimizer_sanity_check] **** You are using ZeRO with an untested optimizer, proceed with caution *****
> [2024-08-12 07:41:07,170] [INFO] [logging.py:96:log_dist] [Rank 0] DeepSpeed Basic Optimizer = FusedAdamW
> [2024-08-12 07:41:07,171] [INFO] [utils.py:56:is_zero_supported_optimizer] Checking ZeRO support for optimizer=FusedAdamW type=<class 'habana_frameworks.torch.hpex.optimizers.FusedAdamW.FusedAdamW'>
> [2024-08-12 07:41:07,171] [WARNING] [engine.py:1196:_do_optimizer_sanity_check] **** You are using ZeRO with an untested optimizer, proceed with caution *****
> 
> 
> ***** train metrics *****
>   epoch                       =         3.0
>   max_memory_allocated (GB)   =       48.81
>   memory_allocated (GB)       =       47.06
>   total_flos                  = 221557680GF
>   total_memory_available (GB) =       93.55
>   train_loss                  =         nan
>   train_runtime               =  0:04:03.07
>   train_samples_per_second    =    7888.147
>   train_steps_per_second      =       1.933
> ```

Hi, as shown in https://github.com/huggingface/optimum-habana/tree/main/examples/contrastive-image-text#deepspeed, --`use_hpu_graphs_for_training` is not for this workload. Please with w/o it.

**Comment by caijimin** [comment]
Thanks Xinyu, it worked after remove  `--use_hpu_graphs_for_training` , and need add keep `--mediapipe_dataloader`.
